### PR TITLE
P#52753 Newsletter activation flag not shown in e-mail

### PR DESCRIPTION
**Pull request by hungnc89**

### CODE REVIEW

**Review by jayay — APPROVED** [review]

File: plugin/FormPostContact.php [hunk 1]
```diff
@@ -182,6 +183,10 @@ private function sendContactRequest(FormData $pFormData, string $recipient = '',
 		if ($this->_pFormPostContactConfiguration->getNewsletterAccepted()) {
```
Comment (jayay) [inline plugin/FormPostContact.php:183]: couldn't this entire block:
```php
		if ($this->_pFormPostContactConfiguration->getNewsletterAccepted()) {
			$requestParams['addressdata']['newsletter_aktiv'] = true;
		}
		if (isset($addressData['newsletter']) && !$this->_pFormPostContactConfiguration->getNewsletterAccepted()) {
			$requestParams['addressdata']['newsletter_aktiv'] = false;
		}
		unset($requestParams['addressdata']['newsletter']);
```

become this?
```php
if (isset($addressData['newsletter'])) {
    $requestParams['addressdata']['newsletter_aktiv'] = $this->_pFormPostContactConfiguration
        ->getNewsletterAccepted();
}
unset($requestParams['addressdata']['newsletter']);
```

**Review by jayay — APPROVED** [review]

**Comment by tung-le-esg** [comment]
> @hungnc89 @tung-le-esg The code doesn't seem to work: Comment by our testers Test with version 2.18.2-3-gab8a22f1 on https://wpptest.onofficeweb.com/:
> 
> The information about the special field "Newsletter" is now completely missing in the mail. No matter whether the field is checked in the contact form or not. See also Contact_Form_Newsletter.png.
> 
> Desired state: The special field "Newsletter" should be listed with "Yes" or "No" in the mail Contact_Form_Newsletter .

@michaelarnold-dev We placed a comment in #2651649 task. We tested it again with the attached plugin file in the task and it works.

**Comment by michaelarnold-dev** [comment]
@tung-le-esg Could you please increase the coverage?

**Comment by tung-le-esg** [comment]
> 

Yes, I will

**Comment by tung-le-esg** [comment]
> @tung-le-esg Could you please increase the coverage?

Yes, I will

**Comment by michaelarnold-dev** [comment]
@tung-le-esg Could you please increase the coverage?

**Comment by tung-le-esg** [comment]
> @tung-le-esg Could you please increase the coverage?

Okay, I will

**Pull request merged**